File: problems/1518-water-bottles/solution.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        res = 0
        full = numBottles
        empty = 0
        while full > 0 or empty >= numExchange:
            res += full
            empty += full
            # for next iter
            full = empty // numExchange
            empty = empty % numExchange

        return res

File: problems/1518-water-bottles/solution.c
int numWaterBottles(int numBottles, int numExchange) {
    int res = 0, full = numBottles, empty = 0;

    while (full > 0 || empty >= numExchange) {
        res += full;
        empty += full;
        full = empty / numExchange;
        empty = empty % numExchange;
    }

    return res;
}